### DISCUSSION TITLE
Possible to have ImageArrayRadius?

### DISCUSSION THREAD

**Discussion by allanepifanio** [Spawnova/ShinsImageScanClass#13] [Ideas]
Hi Sir, Love your work and its very useful, I just want to have a request if its possible for you to implement also:

Scan.ImageArrayRadius
Scan.PixelArrayRadius

Thanks,

**Comment by Spawnova** [comment]
Hi, you can easily search in a radius with the normal ImageArray function like so
```ahk
pointX := 500
pointY := 500
radius := 100

if (Scan.ImageArray("image.png",array)) {
	for k,v in array
	{
		if (GetDistance(v.x,v.y,pointX,pointY) <= radius) {
			msgbox % "image at " v.x "," v.y " is within radius"
		}
	}
}

GetDistance(x1,y1,x2,y2) {
	x:=x1-x2
	y:=y1-y2
	return % sqrt((x*x) + (y*y))
}
```

as far as pixels go I don't currently have an array function for pixels, it's something I could look into adding though,
